### PR TITLE
[ONNX] Fix partial name matching when searching parameter tensors

### PR DESCRIPTION
**Pull request by wschin**
Now we remove name in `onnx_input_names` once it's matched by a parameter so that the same name won't be matched twice.

### CODE REVIEW

File: torch/onnx/_internal/fx/serialization.py [hunk 1]
```diff
@@ -144,12 +144,16 @@ def save_model_with_external_data(
...
             # Step 1 is required because sometimes, tensor names are stored with prefix the dictionary
             # loaded by torch.load.
-            for onnx_input_name in onnx_input_names:
-                if onnx_input_name.endswith(name) or name.endswith(onnx_input_name):
-                    # Find a match. Change name to the matched ONNX input name, so that we
-                    # create initializer with the right ONNX name.
-                    name = onnx_input_name
-                    break
+            if name in onnx_input_names:
+                onnx_input_names.remove(name)
```
Comment (thiagocrepaldi) [inline torch/onnx/_internal/fx/serialization.py:148]:
```suggestion
                # Same input name shouldn't be matched again
                onnx_input_names.remove(name)
```

**Review by thiagocrepaldi — APPROVED** [review]
LGTM

**Comment by BowenBao** [comment]
Is there a potential mismatch situation? Since the need of the PR seems to suggest if an onnx_name is not removed after matched it might get matched to another tensor again.

**Comment by thiagocrepaldi** [comment]
> Is there a potential mismatch situation? Since the need of the PR seems to suggest if an onnx_name is not removed after matched it might get matched to another tensor again.

@wschin FYI

**Pull request closed without merging**